core/src/main/java/org/elasticsearch/common/transport/BoundTransportAddress.java

```diff
@@ -62,18 +65,38 @@ public static BoundTransportAddress readBoundTransportAddress(StreamInput in) th
 
     @Override
     public void readFrom(StreamInput in) throws IOException {
-        boundAddress = TransportAddressSerializers.addressFromStream(in);
+        int boundAddressLength = in.readInt();
+        boundAddresses = new TransportAddress[boundAddressLength];
+        for (int i = 0; i < boundAddressLength; i++) {
+            boundAddresses[i] = TransportAddressSerializers.addressFromStream(in);
+        }
         publishAddress = TransportAddressSerializers.addressFromStream(in);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        TransportAddressSerializers.addressToStream(out, boundAddress);
+        out.writeInt(boundAddresses.length);
+        for (TransportAddress address : boundAddresses) {
+            TransportAddressSerializers.addressToStream(out, address);
+        }
         TransportAddressSerializers.addressToStream(out, publishAddress);
     }
 
     @Override
     public String toString() {
```

Comment:
Is this the piece that gets logged at INFO level on startup? This would fix the confusing printout we do today. 

So we may want to lower the per-address bind printout that we do at INFO level in Netty/HTTP, make it DEBUG or something, because we will be printing them all now at the end?